src_plugins/ppx_deriving_show.cppo.ml

```diff
@@ -12,10 +12,15 @@ open Ast_convenience
 let deriver = "show"
 let raise_errorf = Ppx_deriving.raise_errorf
 
+type options_t = { nofullpath : bool }
+
 let parse_options options =
+  let nofullpath = ref false in
   options |> List.iter (fun (name, expr) ->
     match name with
-    | _ -> raise_errorf ~loc:expr.pexp_loc "%s does not support option %s" deriver name)
+    | "nofullpath" -> nofullpath := Ppx_deriving.Arg.(get_expr ~deriver bool) expr
```

Comment:
I'd prefer to avoid negation here. So, the option would be called `with_path` and default to `true`.